igcollect/ping.py

```diff
@@ -4,83 +4,90 @@
+    parser.add_argument('-p', '--prefix', default='ping',
+                        help='the path to the value in Graphite')
+    parser.add_argument('-t', '--timeout', type=int, default=300,
+                        help='time in milliseconds'
+                             ' till the timeout is retched')
     return parser.parse_args()
 
 
 def main():
     args = parse_args()
-    hostname = args.adress
-    count = args.count
-    prefix = args.prefix
-
-    i = 1
-    hosts = ''
-    for host in hostname:
-        hosts += host + ' '
-        if i % 10 == 0:
-            _thread.start_new_thread(ping, (prefix, hosts, count))
-            hosts = ''
-    if hosts != '':
-        ping(prefix, hosts, count)
-
-    while _thread._count() > 0:
-        sleep(1)
-
-
-def ping(prefix, hosts, count):
-    cmd = ('fping -B1 -q -C {} -p {} {}'.format(count, (20/count)*2000, hosts))
-    fping = subprocess.getoutput(cmd)
-    output = fping.split('\n')
+
+    with Pool(cpu_count() * 2) as p:
+        for i in range(len(args.hosts) % 10):
+            p.starmap(ping, [(args.prefix, args.hosts[i*10:i*10+9],
+                              args.count, args.timeout)])
+        p.starmap(ping, [(args.prefix,
+                          args.hosts[(len(args.hosts) % 10) * 10:
+                                     len(args.hosts) - 1],
+                          args.count, args.timeout)])
+
+
+def ping(prefix, hosts, count, timeout):
+    hosts_string = " ".join(hosts)
+    # -p calculates the time fping waits between single ping probes
+    cmd = ('fping -B1 -q -C {} -p {} -t {} {}'.format(count, (20/count) * 1800,
+                                                      timeout, hosts_string))
+    output = subprocess.getoutput(cmd).split('\n')
     for line in output:
         data = {}
-        part = line.split(':')
-        data['dest'] = part[0].replace(' ', '').replace('.', '_')
-        pings = part[1].split(' ')
-        i = 1
+        parts = line.split(':')
+        data['dest'] = parts[0].replace(' ', '').replace('.', '_')
+        pings = parts[1].split(' ')
+        # first value would be empty
+        pings.pop(0)
+
+        print(pings)
         avg = 0
-        while i <= count:
+        for i, ping in enumerate(pings):
             if pings[i] == '-':
                 data['ping{}'.format(i)] = -1
             else:
                 data['ping{}'.format(i)] = pings[i]
                 avg += float(pings[i])
             i += 1
 
-        avg /= count
-        if avg == 0.0:
-            data['avg'] = '-'
-        else:
-            data['avg'] = avg
-        pings = pings[1:]
+        avg /= len(pings)
+        data['avg'] = avg
+
+        pings = pings[0:]
         data['max'] = max(pings)
         data['min'] = min(pings)
 
         send(prefix, data, count)
 
 
 def send(prefix, data, count):
-    template = prefix + '.{}.{} {} ' + str(int(time()))
+    template = str(prefix) + '.{}.{} {} ' + str(int(time.time()))
 
     fails = 0
```

Comment:
What's the reason you calculate fails in here and not above during the `ping` function?